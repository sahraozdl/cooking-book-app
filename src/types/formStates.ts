import { z } from 'zod';
import { recipeSchema } from '@/app/actions/schema';

export type RecipeSchemaType = z.infer<typeof recipeSchema>;

export type NewRecipeFormState = {
  success: boolean;
  message: string;
  errors?: Record<string, string[]>;
  inputs?: Partial<RecipeSchemaType> & { id?: string };
};
