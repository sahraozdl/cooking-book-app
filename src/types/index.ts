export * from './user';
export * from './options';
export * from './formStates';
export * from './recipeForm';
