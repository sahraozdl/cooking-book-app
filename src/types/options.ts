export interface Category {
  id: string;
  name?: string;
  strCategoryDescription?: string;
  strCategoryThumb?: string;
}

export interface Cuisine {
  id: string;
  name?: string;
  region?: string;
}

export interface Difficulty {
  id: string;
  avgTime?: string;
  name?: string;
}
export interface Serving {
  id: string;
  name?: string;
}
export interface Option {
  id: string;
  name: string;
}
