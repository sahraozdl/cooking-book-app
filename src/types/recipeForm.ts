export interface Ingredient {
  strIngredient: string;
  strMeasure: string;
}
export interface RecipeFormData {
  categories?: { id: string; name: string }[];
  categoryIds?: string[];
  cuisineId?: {
    id: string;
    name: string;
  };
  strMeal: string;
  strInstructions: string;
  strMealThumb: string;
  difficultyId?: { id: string; name: string };
  ingredients: Ingredient[];
  servingsId?: { id: string; name: string };
  authorId: string;
  authorName?: string;
  isAnonymous?: boolean;
  likedBy?: string[];
  savedBy?: string[];
  likeCount?: number;
  saveCount?: number;
  visibility: 'public' | 'private';
  createdAt: {
    seconds: number;
    nanoseconds: number;
  };
}

export interface RecipeWithID extends RecipeFormData {
  id: string;
}
