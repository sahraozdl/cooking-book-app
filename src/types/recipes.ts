export interface NewRecipeFormState {
  success: boolean;
  message: string;
  errors?: {
    [K in keyof RecipeFormData]?: string[];
  };
  inputs?: Partial<RecipeFormData>;
}

export interface Ingredient {
  strIngredient: string;
  strMeasure: string;
}
export interface RecipeFormData {
  categories: string[];
  cuisineId?: string;
  strMeal: string;
  strInstructions: string;
  strMealThumb: string;
  difficultyId?: string;
  ingredients: Ingredient[];
  servingsId: string;
  authorId: string;
  authorName?: string;
  isAnonymous?: boolean;
  likedBy?: string[];
  savedBy?: string[];
  likeCount?: number;
  saveCount?: number;
  visibility: "public" | "private";
  createdAt: {
    seconds: number;
    nanoseconds: number;
  };
}

export interface RecipeWithID extends RecipeFormData {
  id: string;
}

export interface Category {
  id: string;
  name?: string;
  strCategoryDescription?: string;
  strCategoryThumb?: string;
}

export interface Cuisine {
  id: string;
  name?: string;
  region?: string;
}

export interface Difficulty {
  id: string;
  avgTime?: string;
  name?: string;
}
export interface Option {
  id: string;
  name?: string;
}

export interface SearchUser {
  id: string;
  name?: string;
  email?: string;
  photoURL?: string;
}

export interface UserTypes {
  id?: string;
  name?: string;
  email?: string;
  password?: string;
  photoURL?: string;
  createdAt?: Date;
  writes?: string[];
  following?: string[];
  followers?: string[];
  writesCount?: number;
  savedRecipes?: string[];  // ✅
  likedRecipes?: string[];  // ✅
  errors?: {
    name?: string[];
    email?: string[];
    password?: string[];
  };
}
