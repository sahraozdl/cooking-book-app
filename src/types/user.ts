export interface SearchUser {
  id: string;
  name?: string;
  email?: string;
  photoURL?: string;
}

export interface UserTypes {
  id: string;
  name?: string;
  email?: string;
  password?: string;
  photoURL?: string;
  createdAt?: Date;
  writes?: string[];
  following?: string[];
  followers?: string[];
  writesCount?: number;
  savedRecipes?: string[];
  likedRecipes?: string[];
  errors?: {
    name?: string[];
    email?: string[];
    password?: string[];
  };
}
