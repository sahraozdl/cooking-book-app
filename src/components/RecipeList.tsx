import React from "react";
import EntryCard from "./EntryCard";
import { RecipeWithID } from "@/types";

interface RecipeListProps {
  recipes: RecipeWithID[];
  loading?: boolean;
  editable?: boolean;
  onDelete?: (id: string) => void;
  deletingId?: string | null;
}

export default function RecipeList({
  recipes,
  loading = false,
  editable = false,
  onDelete,
  deletingId,
}: RecipeListProps) {
  if (loading) return <p>Loading recipes...</p>;
  if (recipes.length === 0) return <p>No recipes found.</p>;

  return (
    <div className="space-y-4">
      {recipes.map((recipe) => (
        <EntryCard
          key={recipe.id}
          entry={recipe}
          editable={editable}
          onDelete={onDelete ? () => onDelete(recipe.id) : undefined}
          isDeleting={deletingId === recipe.id}
        />
      ))}
    </div>
  );
}
