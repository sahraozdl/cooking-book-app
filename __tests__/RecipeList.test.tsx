import { render, screen } from '@testing-library/react';
import RecipeList from '@/components/RecipeList';
import { RecipeWithID } from '@/types';

jest.mock('@/components/EntryCard', () => {
  return function MockEntryCard({ entry }: { entry: RecipeWithID }) {
    return <div data-testid="entry-card">{entry.strMeal}</div>;
  };
});

const mockRecipes: Partial<RecipeWithID>[] = [
  { id: '1', strMeal: 'Pizza' },
  { id: '2', strMeal: 'Pasta' },
];

describe('RecipeList component', () => {
  it('shows loading state', () => {
    render(<RecipeList recipes={[]} loading />);
    expect(screen.getByText('Loading recipes...')).toBeInTheDocument();
  });

  it('shows empty state', () => {
    render(<RecipeList recipes={[]} />);
    expect(screen.getByText('No recipes found.')).toBeInTheDocument();
  });

  it('renders recipes', () => {
    render(<RecipeList recipes={mockRecipes as RecipeWithID[]} />);
    expect(screen.getAllByTestId('entry-card')).toHaveLength(2);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<RecipeList recipes={mockRecipes as RecipeWithID[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
