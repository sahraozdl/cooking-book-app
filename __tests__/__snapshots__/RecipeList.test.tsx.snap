// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RecipeList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-4"
  >
    <div
      data-testid="entry-card"
    >
      Pizza
    </div>
    <div
      data-testid="entry-card"
    >
      Pasta
    </div>
  </div>
</DocumentFragment>
`;
